models/DataObject/Service.php

```diff
@@ -1819,29 +1820,29 @@ public static function getCsvDataForObject(Concrete $object, $requestedLanguage,
                     $localeService->setLocale($validLanguage);
                     $fieldData = self::getCsvFieldData($currentLocale, $field, $object, $validLanguage, $helperDefinitions);
                     $localizedFieldKey = $field . '-' . $validLanguage;
-                    if (!isset($mappedFieldnames[$localizedFieldKey])) {
-                        $mappedFieldnames[$localizedFieldKey] = $mappedFieldnameBase . '-' . $validLanguage;
+                    if (!isset($mappedFieldnames[$index])) {
+                        $mappedFieldnames[$index] = $mappedFieldnameBase . '-' . $validLanguage;
                     }
-                    $objectData[$localizedFieldKey] = $fieldData;
+                    $objectData[$index] = ['fieldName' => $localizedFieldKey, 'data' => $fieldData];
                 }
 
                 $localeService->setLocale($currentLocale);
             } else {
                 $fieldData = self::getCsvFieldData($requestedLanguage, $field, $object, $requestedLanguage, $helperDefinitions);
-                if (!isset($mappedFieldnames[$field])) {
-                    $mappedFieldnames[$field] = self::mapFieldname($field, $helperDefinitions);
+                if (!isset($mappedFieldnames[$index])) {
+                    $mappedFieldnames[$index] = self::mapFieldname($field, $helperDefinitions);
                 }
 
-                $objectData[$field] = $fieldData;
+                $objectData[$index] = ['fieldName' => $field, 'data' => $fieldData];
             }
+            $index++;
         }
 
         if ($returnMappedFieldNames) {
             $tmp = [];
```

Comment:
This may not be used anymore?

Reply:
Yep! I just added it.

Reply:
The index is needed, but its better to use loop index so I changed it.

Reply:
@cameronfromtorq sorry I meant I don't think `$tmp` is being used anymore.